base/int.jl

```diff
@@ -79,6 +79,13 @@ end
 
 checked_abs(x::Unsigned) = abs(x)
 
+function checked_neg{T<:Signed}(x::T)
+    x == typemin(T) && throw(OverflowError())
```

Comment:
This mirrors existing code in `checked_abs`.

If you prefer, we can replace `Signed` by `Union{Int8,...,Int128}`, leaving the general case undefined. `BigInt` is already handled separately.